### PR TITLE
Allow to send a user picture along with editable fields

### LINKED ISSUE #5029: Allow to send a user picture along with editable fields
In order to upload a user profile picture, we need to change the `editUserAccount()` action (reducer/saga) to optionally take a `picture` parameter and send a form data API call (see #5012).

### PR DESCRIPTION
**Pull request by willdurand**
Fix #5029 – Refs #4276 – ⚠️ Depends on #5013 

---

This PR allows to send a picture file along with editable fields when editing a user profile. Only the second comment is relevant.

### CODE REVIEW

File: src/amo/reducers/users.js [hunk 1]
```diff
@@ -141,15 +142,15 @@ type EditUserAccountAction = {|
...
+  errorHandlerId, picture, userFields, userId,
 }: EditUserAccountParams): EditUserAccountAction => {
   invariant(errorHandlerId, 'errorHandlerId is required');
   invariant(userFields, 'userFields are required');
   invariant(userId, 'userId is required');
 
   return {
     type: EDIT_USER_ACCOUNT,
-    payload: { errorHandlerId, userFields, userId },
+    payload: { errorHandlerId, userFields, picture, userId },
```
Comment (bobsilverberg) [inline src/amo/reducers/users.js]: Nit: I think we try to keep arguments in alphabetical order.

File: tests/unit/amo/api/test_users.js [hunk 2]
```diff
@@ -79,6 +81,30 @@ describe(__filename, () => {
...
+      expectedBody.set('picture_upload', picture);
+
+      mockApi.expects('callApi')
+        .withArgs(sinon.match(({ body }) => {
+          return deepEqual(
+            Array.from(body.entries()),
+            Array.from(expectedBody.entries())
+          );
+        }))
+        .returns(mockResponse());
```
Comment (bobsilverberg) [inline tests/unit/amo/api/test_users.js]: Why is this returning an empty response?

File: src/amo/reducers/users.js (hunk 1 above)
Comment (willdurand) [reply src/amo/reducers/users.js]: Good catch!

File: tests/unit/amo/api/test_users.js (hunk 2 above)
Comment (willdurand) [reply tests/unit/amo/api/test_users.js]: I do not test/verify the response here.

**Pull request merged**